rllib/evaluation/collectors/simple_list_collector.py

```diff
@@ -251,13 +262,12 @@ def add_postprocessed_batch_for_training(
                 training).
         """
         for view_col, data in batch.items():
-            # TODO(ekl) how do we handle this for policies that don't extend
-            # Torch / TF Policy template (no inference of view reqs)?
-            # Skip columns that are not used for training.
-            # if view_col not in view_requirements or \
-            #         not view_requirements[view_col].used_for_training:
-            #     continue
-            self.buffers[view_col].extend(data)
+            # 1) If col is not in view_requirements, we must have a direct
```

Comment:
I think this solves it:
- if - after postprocessing - some column is not in the view-reqs, we must deal with a base-Policy child (w/o auto-view-requirement handling) -> leave as is
- if we do have it in the view reqs AND used_for_training is False -> we must have gone through auto-detection, so it's save to remove it here (this column won't be needed for training).